src/plugins/build.ts

```diff
@@ -22,6 +22,30 @@ export function BuildPlugin(ctx: PWAPluginContext) {
         return injectServiceWorker(html, options, false)
       },
     },
+    async writeBundle() {
+      // add support for new SvelteKit Vite Plugin
+      const {
+        options: {
+          disable,
+          svelteKitVitePluginOptions: {
+            prerenderTimeout = 0,
+            disabled = false,
+          },
+        },
+        viteConfig,
+      } = ctx
+
+      const sveltekitPresent = disabled || disable || !viteConfig.build.ssr
```

Comment:
I would expect `vite-plugin-svelte-kit` to be present only when `!ssr` and so I would expect `sveltekitPresent` to always be falsy

Reply:
We need the plugin on client and on ssr. The client side will use the virtual to inject the workbox-window and the writeBundle will generate the sw and the webmanifest

Reply:
If there is no prerender then the `closeBundle` will work, so the `writeBundle` should not generate the SW. If there is prerender, then `closeBundle` will never be called since there is an `exit` call, so we need to generate the SW on `writeBundle` only on SSR (prerender).

About the `disabled` option (should be called, `usingExternalScript` ;) ), if we are using the `pwa.js` module on the script after build, then we need to exclude from the `writeBundle` since we are generating externally the SW using a new Vite build from `pwa.js` module.

Maybe it will be better to await your changes: I want to give support also for previous versions, what do you think?